plugins/modules/system/parted.py

```diff
@@ -650,6 +683,25 @@ def main():
         if unit and script:
             script = "unit %s %s" % (unit, script)
 
+        # If partition exists, try to resize
+        if part_exists(current_parts, 'num', number) and resize:
```

Comment:
With this change:
```suggestion
        if resize and part_exists(current_parts, 'num', number):
```
no code is run if `resize` is not specified. That definitely makes sure nothing breaks that currently works (and makes me happier to merge this without any maintainer reacting).